ext/openssl/ossl_ssl.c

```diff
@@ -1557,7 +1577,11 @@ ossl_start_ssl(VALUE self, int (*func)(), const char *funcname, VALUE opts)
                 continue;
 #endif
 	    if (errno) rb_sys_fail(funcname);
-	    ossl_raise(eSSLError, "%s SYSCALL returned=%d errno=%d state=%s", funcname, ret2, errno, SSL_state_string_long(ssl));
+        peeraddr_info = peeraddr_ip_str(self);
+        peeraddr = RTEST(peeraddr_info) ? StringValuePtr(peeraddr_info) : "(null)";
+	    ossl_raise(eSSLError, "%s SYSCALL returned=%d errno=%d peeraddr=%s state=%s",
+                funcname, ret2, errno, peeraddr, SSL_state_string_long(ssl));
```

Comment:
`ossl_raise()` is a wrapper around `rb_raise()`, which in turn uses `rb_sprintf()` for formatting. It supports string interpolation like in Ruby code, so this can be written in cleaner way:

```c
ossl_raise(xx, "peeraddr=%"PRIsVALUE" state=...", peeraddr_ip_str(self), [...]);
```

`peeraddr_ip_str()` can handle the error and return the dummy string instead. This is a bit inefficient, but I guess it doesn't matter because it's only an error path inside an error path.